admin/automation/sync.py

```diff
@@ -0,0 +1,231 @@
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def load_credentials(directory):
+    for filename in os.listdir(directory):
+        file_path = os.path.join(directory, filename)
+
+        if os.path.isfile(file_path):
+            with open(file_path, "r") as file:
+                content = file.read().strip()
+                os.environ[filename] = content
+
+
+class Deliverable(str, Enum):
+    SERVICES = "services"
+    EXECUTABLES = "executables"
+    LIBRARIES = "libraries"
+    TESTS = "tests"
+    DEVENV = "development environment"
+
+
+@dataclass
+class Project:
+    name: str
+    branch_name: str = field(init=False)
+    description: str = "\n### Websites"
+    websites: Series = field(default_factory=Series)
+
+    def __post_init__(self):
+        self.branch_name = f"projects/{self.name}"
+
+    def append_websites(self, websites: List[str]):
+        self.websites = pd.concat(
+            [self.websites, pd.Series(websites)], ignore_index=True
+        )
+
+    # TODO: make a template for this
+    def update_description(self):
+        if len(self.websites) > 0:
+            pass
+
+        for site in self.websites:
+            self.description += f"\n- {site}"
+
+    def clean_websites(self):
+        """Remove empty values, clean up URLs and remove duplicates"""
+        self.websites = cleanup_empty(self.websites)
+        self.websites = cleanup_urls(self.websites)
+        self.websites = self.websites.drop_duplicates()
+
+
+class Subgrants(BaseModel):
+    subgrants: List[Subgrant] = []
+
+    def get_websites(self, name: str) -> List[str]:
+        for s in self.subgrants:
+            if name == s.name:
+                return s.websites
+        return []
+
+
+class NotionProject(BaseModel):
+    name: str = ""
+    subgrants: List[str] = []
+
+
+# TODO: get status from Notion
+def main():
+    count = 1
+
+    projects = pd.read_csv(input_file, usecols=["Name", "Subgrants"])
+    projects["Name"] = cleanup_empty(projects["Name"])
+
+    # Some project names have spaces in them, so we'd either need to remove
+    # that in Notion or just replace them here.
+    projects["Name"] = projects["Name"].apply(
+        lambda x: x.replace(" ", "-") if pd.notna(x) else ""
+    )
+
+    # Remove notion URLs from the subgrants. We need these for getting the project websites.
+    projects["Subgrants"] = projects["Subgrants"].apply(
+        lambda x: remove_urls(x).split(" ") if pd.notna(x) else []
+    )
+
+    projects_dict = projects.to_dict(orient="records")
+    projects = [
+        NotionProject(name=item["Name"], subgrants=item["Subgrants"])
+        for item in projects_dict
+    ]
+
+    try:
+        with open(info_file, "r") as f:
+            funds = Subgrants(subgrants=json.load(f))
+    except ValidationError as e:
+        logger.error(f"Failed to parse subgrant information from {info_file}: {e}")
+        exit()
+
+    # TODO: use env variable or argument for repo?
```

Comment:
I recommend a required argument because this easy to get wrong, and where getting it wrong may have undesirable consequences.